crates/goose-cli/src/session/mod.rs

```diff
@@ -398,49 +398,36 @@ impl CliSession {
         editor.set_helper(Some(completer));
 
         let history_file = Paths::state_dir().join("history.txt");
-
-        // Migrate from old location if needed
         let old_history_file = Paths::config_dir().join("history.txt");
-        if old_history_file.exists() && !history_file.exists() {
-            if let Some(parent) = history_file.parent() {
-                if !parent.exists() {
-                    std::fs::create_dir_all(parent)?;
-                }
-            }
-
-            // Try to move the file (rename is atomic on same filesystem)
-            if std::fs::rename(&old_history_file, &history_file).is_err() {
-                // If rename fails (e.g., cross-device), try copy + delete
-                if let Err(copy_err) = std::fs::copy(&old_history_file, &history_file) {
-                    eprintln!("Warning: Failed to migrate history file: {}", copy_err);
-                } else {
-                    // Successfully copied, now try to remove old file
-                    if let Err(remove_err) = std::fs::remove_file(&old_history_file) {
-                        eprintln!(
-                            "Warning: Migrated history file but couldn't remove old file: {}",
-                            remove_err
-                        );
-                    }
-                }
-            }
-        }
 
         if let Some(parent) = history_file.parent() {
             if !parent.exists() {
                 std::fs::create_dir_all(parent)?;
             }
         }
 
-        if history_file.exists() {
+        let history_loaded = if history_file.exists() {
             if let Err(err) = editor.load_history(&history_file) {
                 eprintln!("Warning: Failed to load command history: {}", err);
             }
-        }
+            true
+        } else if old_history_file.exists() {
+            if let Err(err) = editor.load_history(&old_history_file) {
+                eprintln!("Warning: Failed to load command history: {}", err);
+            }
+            true
+        } else {
+            false
+        };
```

Comment:
I'd probably go with something like:

let load_from = [&history_file, &old_history_file]
    .iter()
    .find(|f| f.exists());

if let Some(file) = load_from {
    if let Err(err) = editor.load_history(file) {
        eprintln!("Warning: Failed to load command history: {}", err);
    }
}